### PR TITLE
new structure continue, move utils, move fx macro to re-frame utils n…

### PR DESCRIPTION
**Pull request by flexsurfer**
new structure continue, move utils, move fx macro to re-frame utils

now we have 
`rf/sub, rf/dispartch, rf/defn, rf/merge` in the same namespace

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -640,7 +640,7 @@ SPEC CHECKSUMS:
   FBLazyVector: 352a8ca9bbc8e2f097d680747a8c97ecef12d469
   FBReactNativeSpec: 7dfb84f624136a45727c813ed21d130cd3e61beb
   Folly: b73c3869541e86821df3c387eb0af5f65addfab4
-  glog: 36ce0530c6d2c3a5a4326885ef4069564887a1db
+  glog: 997518ea2aa2d8cd5df9797b641b758d52ecf2bc
```
Comment (ibrkhalil) [inline ios/Podfile.lock]: Why is this updated ?

File: scripts/lint-re-frame-in-quo-components.sh [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 CHANGES=$(git diff --no-ext-diff --diff-filter=d --cached --unified=0 --no-prefix --minimal --exit-code src/quo src/quo2 | grep '^+' || echo '')
 
-INVALID_CHANGES=$(echo "$CHANGES" | grep -E '(re-frame/dispatch|rf/dispatch|re-frame/subscribe|rf/subscribe|rf/sub|<sub|>evt)')
+INVALID_CHANGES=$(echo "$CHANGES" | grep -E '(re-frame/dispatch|rf/dispatch|re-frame/subscribe|rf/subscribe|rf/sub|<sub|>evt|status-im)')
 
 if test -n "$INVALID_CHANGES"; then
-    echo "re-frame dispatch/subscribe is not allowed in quo/quo2 components"
+    echo "re-frame dispatch/subscribe ans status-im are not allowed in quo/quo2 components"
```
Comment (ibrkhalil) [inline scripts/lint-re-frame-in-quo-components.sh]: and*

File: src/utils/collection.cljs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+       (keep-indexed (fn [idx e]
+                       (when (pred e)
+                         idx)))
+       first))
+
+(defn index-by
+  "Given a collection and a unique key function, returns a map that indexes the collection.
+  Similar to group-by except that the map values are single objects (depends on key uniqueness)."
+  [key coll]
+  (into {} (map #(vector (key %) %) coll)))
```
Comment (ibrkhalil) [inline src/utils/collection.cljs]: Do we need a new line at EOF ?

File: src/utils/number.cljs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  Example usage: use it to avoid re-renders caused by floating-point number
+  changes in Reagent atoms. Such numbers can be rounded up to a certain number
+  of `decimal-places` in order to avoid re-rendering due to tiny fractional
+  changes.
+
+  Don't use this function for arbitrary-precision arithmetic."
+  [n decimal-places]
+  (let [scale (Math/pow 10 decimal-places)]
+    (/ (Math/round (* n scale))
+       scale)))
```
Comment (ibrkhalil) [inline src/utils/number.cljs]: Do we need a new line at EOF ?

File: src/utils/string.cljs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+            max-len   (- threshold 3)
+            start-len (Math/ceil (/ max-len 2))
+            end-len   (Math/floor (/ max-len 2))
+            start     (subs s 0 start-len)
+            end       (subs s (- str-len end-len) str-len)]
+        (str start "..." end))
+      (str (subs s 0 (- threshold 3)) "..."))
+    s))
+
+(def truncate-str (memoize truncate-str-memo))
```
Comment (ibrkhalil) [inline src/utils/string.cljs]: Do we need a new line at EOF ?

**Review by ibrkhalil — APPROVED** [review]
Kindly revert changes to podfile

File: ios/Podfile.lock (hunk 1 above)
Comment (flexsurfer) [reply ios/Podfile.lock]: don't know, reverted

**Pull request merged**